sdk/src/main/scala/io/horizen/account/api/http/route/AccountTransactionApiRoute.scala

```diff
@@ -538,15 +566,89 @@ case class AccountTransactionApiRoute(override val settings: RESTApiSettings,
     }
   }
 
+  def registerForger: Route = (post & path("registerForger")) {
+    withBasicAuth {
+      _ => {
+        entity(as[ReqRegisterForger]) { body =>
+          // lock the view and try to create CoreTransaction
+          applyOnNodeView { sidechainNodeView =>
+            val valueInWei = ZenWeiConverter.convertZenniesToWei(body.stakedAmount)
+
+            // default gas related params
+            val baseFee = sidechainNodeView.getNodeState.getNextBaseFee
+            var maxPriorityFeePerGas = BigInteger.valueOf(120)
+            var maxFeePerGas = BigInteger.TWO.multiply(baseFee).add(maxPriorityFeePerGas)
+            var gasLimit = BigInteger.valueOf(500000)
+
+            if (body.gasInfo.isDefined) {
+              maxFeePerGas = body.gasInfo.get.maxFeePerGas
+              maxPriorityFeePerGas = body.gasInfo.get.maxPriorityFeePerGas
+              gasLimit = body.gasInfo.get.gasLimit
+            }
+            val txCost = valueInWei.add(maxFeePerGas.multiply(gasLimit))
+
+            val secret = getFittingSecret(sidechainNodeView, None, txCost)
+
+            secret match {
+              case Some(secret) =>
+
+                val nonce = body.nonce.getOrElse(sidechainNodeView.getNodeState.getNonce(secret.publicImage.address))
+
+                val smartContractAddressStr = body.smartcontractAddress.getOrElse(NULL_ADDRESS_WITH_PREFIX_HEX_STRING)
```

Comment:
We should check that input params are correct (e.g. value is >= minStake) before creating and sending the transaction, so the sender can save gas in case or errors.
Same for checking on ForgerStake V2 activation.

Reply:
That is now implemented by you thanks!